primitives/src/big_num.rs

```diff
@@ -7,8 +7,62 @@ use std::str::FromStr;
 use num::rational::Ratio;
 use num::{BigUint, CheckedSub, Integer};
 use num_derive::{Num, NumOps, One, Zero};
+use num_traits::Pow;
 use serde::{Deserialize, Deserializer, Serialize, Serializer};
 
+/// DAI has precision of 18 decimals
+/// For CPM we have 3 decimals precision, but that's for 1000 (3 decimals more)
+/// This in terms means we need 18 - (3 + 3) = 12 decimals precision
+pub const GLOBAL_MULTIPLIER: Multiplier = Multiplier(12);
+
+#[derive(Serialize, Deserialize, Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+/// Multiplier - Pow of 10 (10**n)
+pub struct Multiplier(u64);
+
+impl Mul<PrecisionU64> for Multiplier {
+    type Output = BigUint;
+
+    fn mul(self, rhs: PrecisionU64) -> Self::Output {
+        let real_multiplier = BigUint::from(10u8).pow(BigUint::from(GLOBAL_MULTIPLIER.0));
+
+        real_multiplier * rhs.0
+    }
+}
+
+impl Into<BigNum> for Multiplier {
+    fn into(self) -> BigNum {
+        BigNum(self.into())
+    }
+}
+
+impl Into<BigUint> for Multiplier {
+    fn into(self) -> BigUint {
+        BigUint::from(10u8).pow(BigUint::from(self.0))
+    }
+}
+
+///
+// @TODO: (De)serialize
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct PrecisionU64(u64);
+
+impl Into<BigNum> for PrecisionU64 {
+    fn into(self) -> BigNum {
+        BigNum(GLOBAL_MULTIPLIER * self)
+    }
+}
+
+impl From<BigNum> for PrecisionU64 {
```

Comment:
Should we use `From` with default of `0` or should we impl `TryFrom` and fail if this conversion cannot be done?

Reply:
TryFrom would be better as 0 may be magic :( I think error should be handled properly